src/librustc_mir/borrow_check/mod.rs

```diff
@@ -1294,6 +1300,9 @@ impl<'cx, 'gcx, 'tcx> MirBorrowckCtxt<'cx, 'gcx, 'tcx> {
         match *rvalue {
             Rvalue::Ref(_ /*rgn*/, bk, ref place) => {
                 let access_kind = match bk {
+                    BorrowKind::Shallow => {
+                        (Shallow(Some(ArtificialField::Discriminant)), Read(ReadKind::Borrow(bk)))
```

Comment:
From looking over the code, it seems like the effect of this is that if you have a borrow of (say) `a.b.c` and we shallow borrow `a.b`, this won't be considered a conflict. I think .. this is correct, but I'm wondering if maybe we should add another `ArtificialField` variant, like `ShallowBorrow`. Re-using `Discriminant` seems to give bad "intuitions" to me, though I guess I can't think where it would go wrong.

Maybe we should just rename `ArtificialField::Discriminant` to `ShallowBorrow`, actually.